index.d.ts

```diff
@@ -0,0 +1,28 @@
+import { Linter, Rule } from 'eslint';
+
+declare const plugin: {
+  meta: {
+    name: string;
+    version: string;
+  };
+  environments: {
+    globals: {
+      globals: {
+        [key: string]: boolean;
+      };
+    };
+  };
+  configs: {
+    all: Linter.Config;
+    recommended: Linter.Config;
+    style: Linter.Config;
+    'flat/all': Linter.FlatConfig;
+    'flat/recommended': Linter.FlatConfig;
+    'flat/style': Linter.FlatConfig;
+  };
+  rules: {
+    [key: string]: Rule.RuleModule;
+  };
+};
+
+export default plugin;
```

Comment:
issue: this is the correct export type for how the package is published:

```suggestion
export = plugin;
```